packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -133,13 +133,19 @@ abstract class RendererBinding extends BindingBase implements SchedulerBinding,
   ///
   /// Called automatically when the binding is created.
   void initSemantics() {
-    SemanticsNode.onSemanticsEnabled = renderView.scheduleInitialSemantics;
     shell.provideService(mojom.SemanticsServer.serviceName, (core.MojoMessagePipeEndpoint endpoint) {
+      ensureSemantics();
       mojom.SemanticsServerStub server = new mojom.SemanticsServerStub.fromEndpoint(endpoint);
-      server.impl = new SemanticsServer();
+      server.impl = new SemanticsServer(semanticsOwner: pipelineOwner.semanticsOwner);
     });
   }
 
+  void ensureSemantics() {
+    if (pipelineOwner.semanticsOwner == null)
+      renderView.scheduleInitialSemantics();
```

Comment:
o_O

Not sure what to suggest instead. It's a bit weird to rely so explicitly on one object changing another object and having a third object be responsible for making sure the first gets called when the second needs it...